piquasso/_backends/fock/general/calculations.py

```diff
@@ -287,3 +291,69 @@ def _add_occupation_number_basis(
     dual_index = state._space.index(bra)
 
     state._density_matrix[index, dual_index] = coefficient
+
+
+def measure_homodyne(state: FockState, instruction: Instruction, shots: int) -> Result:
```

Comment:
This function is almost exactly the same as the one for the pure one. Try not to repeat yourself.